locksmithctl/daemon.go

```diff
@@ -309,8 +313,15 @@ func runDaemon() int {
 		strategy = StrategyBestEffort
 	}
 
+	// XXX: complain loudly if besteffort is used
+	if strategy == StrategyBestEffort {
+		dlog.Errorf("Reboot strategy %q is deprecated and will be removed in the future.", strategy)
+		dlog.Errorf("Please explicitly set the reboot strategy to one of %v", []string{StrategyOff, StrategyReboot, StrategyEtcdLock})
+		dlog.Error("See https://coreos.com/os/docs/latest/update-strategies.html for details on configuring reboot strategies.")
+	}
+
 	if strategy == StrategyOff {
-		fmt.Fprintf(os.Stderr, "Reboot strategy is %q - shutting down.\n", strategy)
+		dlog.Warningf("Reboot strategy is %q - shutting down.", strategy)
```

Comment:
Perfect, that means anyone configuring locksmith with cloudinit is going to get our "OMG DON"T USE best-effor" error on first boot even though locksmith shortly afterwards restarts with the right settings. *sigh*